### PR TITLE
feat(auth): Support generate oob code request type VERIFY_AND_CHANGE_EMAIL

### PR DESCRIPTION
**Pull request by Xiaoshouzi-gh**
Support generate email link for VERIFY_AND_CHANGE_EMAIL action.

### Discussion

  * Read the contribution guidelines (CONTRIBUTING.md).
  * If this has been discussed in an issue, make sure to link to the issue here. 
    If not, go file an issue about this **before creating a pull request** to discuss.

### Testing

  * Make sure all existing tests in the repository pass after your change.
  * If you fixed a bug or added a feature, add a new test to cover your code.

### API Changes

  * At this time we cannot accept changes that affect the public API.  If you'd like to help 
    us make Firebase APIs better, please propose your change in an issue so that we 
    can discuss it together.

### CODE REVIEW

File: src/auth/base-auth.ts
Comment (renkelvin) [inline src/auth/base-auth.ts:837]: address

**Review by renkelvin — APPROVED** [review]
LGTM

**Review by lahirumaramba — APPROVED** [review]
LGTM! Left a few comments. Thanks!

**Comment by lahirumaramba** [comment]
Please get TW's approval on the reference docs before you merge this. Thank you!

File: src/auth/base-auth.ts
Comment (kevinthecheung) [inline src/auth/base-auth.ts:831]:
```suggestion
   * Generates an out-of-band email action link to verify the user's ownership
```

File: src/auth/base-auth.ts
Comment (kevinthecheung) [inline src/auth/base-auth.ts:847]:
```suggestion
   *     The state URL provided must belong to a domain that is authorized
   *     in the console, or an error will be thrown.
```

**Review by kevinthecheung — APPROVED** [review]

File: test/unit/auth/auth-api-request.spec.ts
Comment (Xiaoshouzi-gh) [reply test/unit/auth/auth-api-request.spec.ts]: Adopted the 2nd method and merge the tests together.  Kept the variable as `let` since there are changes in the existing logic on the variable.

File: src/auth/auth-api-request.ts
Comment (Xiaoshouzi-gh) [reply src/auth/auth-api-request.ts]: Done.

File: src/auth/base-auth.ts
Comment (Xiaoshouzi-gh) [reply src/auth/base-auth.ts]: Fixed both.  Thanks Kelvin!

File: src/auth/base-auth.ts
Comment (Xiaoshouzi-gh) [reply src/auth/base-auth.ts:837]: Kept it as email account as the previous public method (verify_email, reset_password etc) uses the same documentation on email paramater.  Let me know if you feel strong about `email address`.

**Pull request merged**
